apple/internal/templates/apple_simulator.template.py

```diff
@@ -444,37 +454,71 @@ def temporary_simulator(
               prefix="com.apple.CoreSimulator.SimRuntime",
               runtime_platform=runtime_platform,
               runtime_version_name=runtime_version_name,
-          )
+          ),
       ],
       encoding="utf-8",
       check=True,
-      stdout=subprocess.PIPE)
+      stdout=subprocess.PIPE,
+  )
   udid = simctl_create_result.stdout.rstrip()
   try:
     logger.info("Killing all running simulators...")
-    subprocess.run(["pkill", "Simulator"],
-                   stderr=subprocess.DEVNULL,
-                   check=False)
+    subprocess.run(
+        ["pkill", "Simulator"], stderr=subprocess.DEVNULL, check=False
+    )
     yield udid
   finally:
     logger.info("Shutting down simulator with udid: %s", udid)
-    subprocess.run([simctl_path, "shutdown", udid],
-                   stderr=subprocess.DEVNULL,
-                   check=False)
+    subprocess.run(
+        [simctl_path, "shutdown", udid], stderr=subprocess.DEVNULL, check=False
+    )
     logger.info("Deleting simulator with udid: %s", udid)
     subprocess.run([simctl_path, "delete", udid], check=True)
 
 
+def register_dsyms(dsyms_dir: str):
+  """Adds all dSYMs in `dsyms_dir` to the symbolscache.
+
+  Args:
+    dsyms_dir: Path to directory potentially containing dSYMs
+  """
+  symbolscache_command = [
+      "/usr/bin/symbolscache",
```

Comment:
Interesting. We might want to do something like this in rules_xcodeproj.

Reply:
theoretically shouldn't be needed, but some folks at wwdc also recommended this to me when instruments didn't symbolicate things